selectrum.el

```diff
@@ -1637,12 +1641,9 @@ listed candidates (so, for example,
 HISTORY is the `minibuffer-history-variable' to use (by default
 `minibuffer-history').
 
-NO-MOVE-DEFAULT-CANDIDATE, if non-nil, means that the default
-candidate is not sorted first. Instead, it is left at its
-original position in the candidate list. However, it is still
-selected initially. This is handy for `switch-to-buffer' and
-friends, for which getting the candidate list out of order at all
-is very confusing.
+NO-MOVE-DEFAULT-CANDIDATE is ignored and only kept for backward
```

Comment:
I still wonder about this change. Why not just keep the current behavior?

My suggestion would be the following:

1. Expose all the functionality to want to make customizable via variables. Variables are your preferred way and I think now that it is okay! At least when something like #322 is under consideration.
2. Make selectrum--read private and deprecate selectrum-read in favor of selectrum-completing-read
3. Refactor selectrum--read as you see fit :)

Reply:
I did it this way because I want the user option to have precedencoe. But I'm starting to think maybe we should just way with this and make selectrum--read private first so we do it all in one go.